vllm/sequence.py

```diff
@@ -489,6 +490,17 @@ def __repr__(self) -> str:
                 f"num_blocks={self.n_blocks}, ")
 
 
+@dataclass
+class SequenceGroupState:
+    """Mutable state tied to a specific sequence group"""
+
+    # for multi-step decoding
+    num_lookahead_slots: int = 0
+    num_steps: int = 1
+    remaining_steps: int = 0
+    current_step: int = 0
```

Comment:
Can these 2 fields be combined? Intuitively `current_step + remaining_steps` should always be `num_steps`?

Reply:
@comaniac `current_step` and `num_steps` would be sufficient in other words?

Reply:
Yeah that's what I want to confirm

Reply:
[22daa7e](https://github.com/vllm-project/vllm/pull/7000/commits/22daa7ed840a03f12ff89d933c7035176bd8ba1d)